src/main.cpp

```diff
@@ -1384,6 +1453,11 @@ bool CheckTransactionWithoutProofVerification(const CTransaction& tx, CValidatio
                              REJECT_INVALID, "bad-cb-has-spend-description");
         // See ContextualCheckTransaction for consensus rules on coinbase output descriptions.
 
+        // A coinbase transaction cannot have TZE inputs
+        if (tx.vtzein.size() > 0)
```

Comment:
I don't think coinbases should have TZE outputs either. If they're allowed to have those, miners could potentially sneak money through TZEs to avoid having to shield them first.